packages/ra-ui-materialui/src/button/DeleteWithConfirmButton.tsx

```diff
@@ -36,8 +36,8 @@ const DeleteWithConfirmButton: FC<DeleteWithConfirmButtonProps> = props => {
         icon = defaultIcon,
         label = 'ra.action.delete',
         onClick,
-        record,
-        resource,
+        record = { id: '' },
```

Comment:
Why? The record is sometimes undefined, in which case we should not display the button (or allow it to be clicked). It doesn't make sense to delete a record with id ''

Reply:
I was using type-script with strictMode = true. I'll change it.